src/lib/Components/Bidding/Components/Input.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { Component } from "react"
+import { TextInputProperties } from "react-native"
+import { Input as StyledInput, InputProps as StyledInputProps } from "../Elements/Input"
+
+interface InputProps extends StyledInputProps, TextInputProperties {
+  error?: boolean
+}
+
+class InputStete {
+  borderColor: "black10" | "purple100" | "red100"
+}
+
+export class Input extends Component<InputProps, InputStete> {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      borderColor: this.props.error ? "red100" : "black10",
+    }
+  }
+
+  onBlur() {
+    this.setState({
+      borderColor: this.props.error ? "red100" : "black10",
+    })
+  }
+
+  onFocus() {
+    this.setState({ borderColor: "purple100" })
+  }
+
+  render() {
+    return (
+      <StyledInput
+        border={1}
+        borderColor={this.state.borderColor}
+        fontSize={3}
+        onBlur={() => this.onBlur()}
+        onFocus={() => this.onFocus()}
+        p={3}
+        pb={2}
+        {...this.props}
+      />
+    )
+  }
+}
```

Comment:
With this new component, could we add a few tests? It's got state and behaviour that we should have covered. I know the "best way" for Emission to be tested is still undecided, but even snapshot tests would be better than nothing.

Reply:
Following up on this comment, in `Input-tests.tsx` I think we should add tests for different border colours and the presence of an error and for focusing.

Reply:
It always sounds good to say "Add more tests", but sometimes it's a trap. I don't find snapshot testing very useful, and tests that are not helpful are liabilities we should have less, so I added more explicit tests. Even these tests could be liabilities because they depend on implementation details (`onBlur` and `onFocus`).

Reply:
Okay, thanks. I'll take a look at the tests. I agree that snapshot tests aren't _that_ useful, but I have to stress that Emission's test coverage is pretty poor right now. The Purchase team has the opportunity to lead by example, to figure out what a meaningful test looks like for this type of component, etc. This is part of the work I've done gathering examples here: https://github.com/artsy/emission/pull/1034/files#diff-b5e6d39e5fb762a421d5bc45ebba4a0eR37